### PR TITLE
Wfh/eval max concurrency

### PR DESCRIPTION
**Pull request by hinthornw**

### CODE REVIEW

File: libs/langchain/langchain/callbacks/tracers/evaluation.py [hunk 1]
```diff
@@ -81,7 +82,13 @@ def __init__(
         )
         self.client = client or langchain_tracer.get_client()
         self.evaluators = evaluators
-        self.executor = _get_executor()
+        if max_concurrency is None:
+            self.executor = _get_executor()
+        elif max_concurrency > 0:
+            self.executor = ThreadPoolExecutor(max_workers=max_concurrency)
+            weakref.finalize(self, self.executor.shutdown, False)
```
Comment (eyurtsev) [inline libs/langchain/langchain/callbacks/tracers/evaluation.py:89]: is there a way to pass the `False` as a named argument?

File: libs/langchain/langchain/callbacks/tracers/evaluation.py [hunk 2]
```diff
@@ -133,9 +140,12 @@ def _persist_run(self, run: Run) -> None:
...
         run_.reference_example_id = self.example_id
         for evaluator in self.evaluators:
-            self.futures.add(
-                self.executor.submit(self._evaluate_in_project, run_, evaluator)
-            )
+            if self.executor is None:
+                self._evaluate_in_project(run_, evaluator)
+            else:
+                self.futures.add(
+                    self.executor.submit(self._evaluate_in_project, run_, evaluator)
```
Comment (eyurtsev) [inline libs/langchain/langchain/callbacks/tracers/evaluation.py:147]: Should the results be written to something that's thread safe? looks liek results end up in logged_eval_results default dict w/ append i don't really have a sense of whether .append is atomic

Comment (hinthornw) [reply libs/langchain/langchain/callbacks/tracers/evaluation.py:147]: Oh interesting ya I'll check

File: libs/langchain/langchain/callbacks/tracers/evaluation.py (hunk 1 above)
Comment (hinthornw) [reply libs/langchain/langchain/callbacks/tracers/evaluation.py:89]: Could use a lambda ya

Comment (hinthornw) [reply libs/langchain/langchain/callbacks/tracers/evaluation.py]: Actually should have it be true

**Pull request merged**